### PR TITLE
Turn the Baggage metadata into a simple wrapper for a String.

### PR DESCRIPTION
**Pull request by jkwatson**
This is the first step toward a W3C Baggage Propagator implementation.

Amends #1530

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/baggage/Entry.java [hunk 1]
```diff
@@ -71,11 +66,12 @@ public static Entry create(String key, String value, EntryMetadata entryMetadata
   public abstract String getValue();
 
   /**
-   * Returns the {@link EntryMetadata} associated with this {@link Entry}.
+   * Returns the (optional) {@link EntryMetadata} associated with this {@link Entry}.
```
Comment (anuraaga) [inline api/src/main/java/io/opentelemetry/baggage/Entry.java:69]:
```suggestion
   * Returns the {@link EntryMetadata} associated with this {@link Entry}, or {@code null} if there is no metadata.
```

**Review by anuraaga — APPROVED** [review]

File: api/src/main/java/io/opentelemetry/baggage/Entry.java (hunk 1 above)
Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/baggage/Entry.java:69]: I've been preferring `EntryMetadata.EMPTY` rather than null. Did I miss a spot?

Comment (anuraaga) [reply api/src/main/java/io/opentelemetry/baggage/Entry.java:69]: Oh that's fine too, in that case can remove `@Nullable` I think - I just noticed the annotation but didn't look at the code

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/baggage/Entry.java:69]: oh yeah. I should remove that. will do!

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/baggage/Entry.java:68]: done

**Review by anuraaga — APPROVED** [review]

**Comment by jkwatson** [comment]
Merging, since #1767 which stacks on this PR has 3 approvals. :)

**Pull request merged**